.github/workflows/build.yml

```diff
@@ -0,0 +1,44 @@
+name: Build
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: '0 3 * * *'
+  workflow_dispatch:
+
+jobs:
+  Sonar-Check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup Java
+        uses: actions/setup-java@v4
+        with:
+          java-version: '11'
```

Comment:
Sonar deprecated running on Java 11, we should use at least 17, ideally 21 (to be aligned with all other repos)